### DISCUSSION TITLE
How to configure triggered of the end of a navigation?

### DISCUSSION THREAD

**Discussion by ChristopherPHolder** [GoogleChrome/lighthouse#15898] [General]
How can I configure a navigation audit to end at a specific time? 

Lets say i have an application that when you navigate to it, it loads the page and continuously updates the content inside of it.

How would i be able to do something like this?

```ts
  await flow.startNavigation();

  await page.goto(homeUrl, { waitUntil: 'load' });
  await page.waitForSelector('After this is here i consider navigation as ended');

  await flow.endNavigation();
```

The issue is that after that moment were the content is loaded a bunch of things are still happening. 

I receive in information from a socket and condinously update the date so it does not really consider the site has finished loading and i get this type of message: 

![image](https://github.com/GoogleChrome/lighthouse/assets/40126819/16337a7a-eda7-49a7-888d-5b60a4a52e03)

This is an example url where i have experiences this issue: https://betjack.com

**Comment by adamraine** [comment]
You can't configure a specific time because there are certain conditions (e.g. network idle) that Lighthouse needs before it can finish auditing the page.

Lighthouse navigation mode will always wait for network idle before finishing. So [`flow.endNavigation`](https://github.com/GoogleChrome/lighthouse/blob/main/docs/user-flows.md#navigations-in-the-nodejs-api) will still wait for network idle even you call it before the quiet period starts. There isn't really a way to configure when Lighthouse considers the network idle.

FWIW the inflight requests on your page are considered critical requests by Lighthouse. Lighthouse considers the network to be idle when there are 0 inflight critical requests AND <=2 inflight non-critical requests. So if you lower the priority of those requests (using something like [`fetchpriority`](https://web.dev/articles/fetch-priority)) this should allow for Lighthouse to finish in the time limit.
